### PR TITLE
Submit com.tomjwatson.Emote

### PR DESCRIPTION
**Pull request by vemonet**
Submission of Emote, a GTK based emoji picker, to flathub

### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. 

**Link:** @tom-james-watson agreed in issue https://github.com/tom-james-watson/Emote/pull/100

- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned.

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: com.tomjwatson.Emote.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+app-id: com.tomjwatson.Emote
+runtime: org.gnome.Platform
+runtime-version: "43"
+sdk: org.gnome.Sdk
+command: emote
+finish-args:
+  - --share=ipc
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --filesystem=xdg-config/autostart:create
```
Comment (hfiguiere) [inline com.tomjwatson.Emote.yml:10]: You should use the portals for that.

File: com.tomjwatson.Emote.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          project-id: 49082
+          url-template: https://github.com/bugaevc/wl-clipboard/archive/v$version.tar.gz
+    cleanup:
+      - /share/man
+      - /share/bash-completion
+  - name: emote
+    buildsystem: meson
+    sources:
+      - type: dir
+        path: "./"
```
Comment (hfiguiere) [inline com.tomjwatson.Emote.yml:65]: this will not work. There is no source.

Comment (vemonet) [reply com.tomjwatson.Emote.yml]: Indeed, I forgot to change the source to the git repository, I fixed it here: https://github.com/flathub/flathub/pull/4265/commits/38f1e8d2384e216a9ca2da1f78c29afd586cacd3

**Comment by hfiguiere** [comment]
bot, build com.tomjwatson.Emote

File: com.tomjwatson.Emote.yml (hunk 1 above)
Comment (hfiguiere) [reply com.tomjwatson.Emote.yml:10]: This is why the build failed. Note that this is code that needs to be written. not "something in the manifest"

File: com.tomjwatson.Emote.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      - type: archive
+        url: https://github.com/kupferlauncher/keybinder/releases/download/keybinder-3.0-v0.3.2/keybinder-3.0-0.3.2.tar.gz
+        sha256: e6e3de4e1f3b201814a956ab8f16dfc8a262db1937ff1eee4d855365398c6020
+  - name: xdotool
+    no-autogen: true
+    make-args:
+      - WITHOUT_RPATH_FIX=1
+    make-install-args:
+      - PREFIX=${FLATPAK_DEST}
+      - INSTALLMAN=${FLATPAK_DEST}/share/man
```
Comment (hfiguiere) [inline com.tomjwatson.Emote.yml:35]: you should actually cleanup the man.

Comment (vemonet) [reply com.tomjwatson.Emote.yml]: done normally!

File: com.tomjwatson.Emote.yml (hunk 1 above)
Comment (vemonet) [reply com.tomjwatson.Emote.yml:10]: No problem if there is a bit of code to be written, but is there any example or documentation about how we can enable a flatpak package to start automatically on boot in the background using the portal API?

Comment (vemonet) [reply com.tomjwatson.Emote.yml:10]: Found something that works:

```python
import dbus
bus = dbus.SessionBus()
obj = bus.get_object("org.freedesktop.portal.Desktop", "/org/freedesktop/portal/desktop")
inter = dbus.Interface(obj, "org.freedesktop.portal.Background")
res = inter.RequestBackground('', {
    'reason': 'Emote autostart',
    'autostart': True, 'background': True,
    'commandline': dbus.Array(['emote'])
})
```

Will push the fix to try a new build asap

Comment (vemonet) [reply com.tomjwatson.Emote.yml]: I just pushed the fix without the filesystem permission and using the portal Background API, thanks for the recommendation @hfiguiere

**Comment by tom-james-watson** [comment]
Thanks for your reviews so far, @hfiguiere - appreciate it!

```
• style-invalid         : Content before <ul> is too short [13], at least 20 characters required
```

Can you help decipher what the issue is here? Is it that it's complaining that there is not enough content in https://github.com/tom-james-watson/Emote/blob/master/flatpak/com.tomjwatson.Emote.metainfo.xml before a `<ul>` is used? I'm not sure what "13" refers to?

**Comment by vemonet** [comment]
I checked our appdata and this is the `<description>` we have:

```html
  <description>
    <p>
      Emote is a modern emoji picker for Linux 🚀
    </p>
    <p>
      Written in GTK3, Emote is lightweight and stays out of your way.
    </p>
    <p>
      Launch the emoji picker with the configurable keyboard shortcut <code>Ctrl+Alt+E</code>, and select one or more emojis to paste them into the currently focused app.
    </p>
    <ul style="list-style: none;">
      <li>🍾 Built as a popup: quick invocation, and disappears when not needed, does not stay as a standalone window</li>
      <li>🫠 Provide a large and up-to-date list of emojis retrieved from <a href="https://openmoji.org" target="_blank">openmoji.org</a></li>
      <li>🧠 Shows the last used emojis by default</li>
      <li>🔎 Search text box automatically focused and ready to type when invoked</li>
      <li>⌨️ Can use shortcuts to navigate and select emojis</li>
      <li>✒️ Selected emoji automatically pasted to your currently focused app (on X11 only)</li>
    </ul>
    <p>
      ℹ️ Notes:
    </p>
    <ul style="list-style: none;">
      <li>⚡️ Emote <a href="https://github.com/tom-james-watson/Emote/issues/54" target="_blank">shows up faster</a> when invoked using the built-in keyboard shortcut (<code>Ctrl+Alt+E</code> by default), than when using a manually registered keyboard shortcut.</li>
      <li>🪟 Emote under Wayland cannot automatically paste the emoji into other apps, and also requires manual registering of a global keyboard shortcut - <a href="https://github.com/tom-james-watson/Emote/wiki/Hotkey-In-Wayland" target="_blank">Hotkey In Wayland</a>. This is due to intentional restrictions in the design of Wayland itself.</li>
    </ul>
  </description>
```

There is a lot more than 20 char before the first `<ul>`, maybe we need more characters between the 2 `<ul>`?

**Comment by tom-james-watson** [comment]
Ah, yeah that might be it. Pushed https://github.com/tom-james-watson/Emote/commit/4bb69c4cac879bc75731c5533ad1893396df6e4f to master which should address that if it's the problem. Can you update the commit ref please @vemonet?

**Comment by hfiguiere** [comment]
first `style` is NOT allowed. This is not HTML.

**Comment by tom-james-watson** [comment]
Removed style in https://github.com/tom-james-watson/Emote/commit/daf8b64749097e7bd883191d21b31281d40c3c30

**Comment by vemonet** [comment]
I updated the commit reference after removing some more HTML elements (e.g <a> and <code>)

**Comment by tom-james-watson** [comment]
bot, build com.tomjwatson.Emote

**Comment by tom-james-watson** [comment]
Build installs and runs fine, but the .desktop file doesn't work. I think https://github.com/tom-james-watson/Emote/pull/104 should fix it.

**Comment by vemonet** [comment]
bot, build com.tomjwatson.Emote

**Comment by vemonet** [comment]
bot, build com.tomjwatson.Emote

**Pull request closed without merging**